Branch-SDK/Branch-SDK/BNCPreferenceHelper.m

```diff
@@ -181,15 +181,7 @@ + (NSInteger)getRetryCount {
 }
 
 + (NSString *)getAppKey {
-    NSString *ret = [[[NSBundle mainBundle] infoDictionary] objectForKey:KEY_APP_KEY];
-    if (!ret || ret.length == 0) {
-        // for backward compatibility
-        ret = [BNCPreferenceHelper readStringFromDefaults:KEY_APP_KEY];
```

Comment:
Yes, I agree. So what you are expecting is our SDK should not work with either old users or new users in this case (no key is specified).
What this line was meant to do was to "save" the old user when the developer forgets to put in the key (either branch key or app key), i.e. at least those old users could still use the SDK correctly. Guess this would be a rare case, but just want to have some backup.